### PR TITLE
Simplify interaction with builder

### PR DESCRIPTION
**Pull request by jbearer**
* Remove block building parameters (`min_transactions`, `max_transactions`, `propose_min_round_time`, `propose_max_round_time`) from HotShot, let the builder deal with that
* Loop until we successfully get blocks (or time out), but no more than that; take a block from the first set that we receive
* Choose block by max fee per byte, not max total fee
* Apply timeout to entire builder interaction, not just `get_available_blocks`

### CODE REVIEW

File: config/ValidatorConfigOutput
Comment (ss-es) [inline config/ValidatorConfigOutput:1]: I'm not sure what this file is or what this change is

cc: @mrain I think you were the last person to edit this file

**Review by ss-es — APPROVED** [review]
Makes sense to me and it appears the examples are still committing a normal amount of transactions in CI, so I think this should be fine

Approving, but I'm not that familiar with the builder (if you wanted to wait for a more careful review from someone else)

File: config/ValidatorConfigOutput
Comment (jbearer) [reply config/ValidatorConfigOutput:1]: Huh I also don't know what this is or where this came from. Definitely didn't mean to change it

Comment (jbearer) [reply config/ValidatorConfigOutput:1]: It's in the .gitignore so maybe it got accidentally committed at some point?

File: crates/hotshot/src/lib.rs
Comment (nyospe) [inline crates/hotshot/src/lib.rs:693]: H-uh. I guess I should've checked whether someone else was doing this.

File: crates/task-impls/src/transactions.rs
Comment (nyospe) [inline crates/task-impls/src/transactions.rs:264]: Ohhhh... so, okay, this doesn't outright make things worse, but... it doesn't make them better either.

So, the implementation here basically took what the builder API was doing, said, "eh, let's ignore all of this and why it was designed this way, and take the most reductive and least efficient approach we can manage". And now that's been tied up with a bow into a nice opaque function.

We have three API calls.

The first should allow HotShot to query any number of builders for their available blocks, and (assuming we ever factor size) wait some amount of time to collect all results that arrive, and choose between the offerings based on fee and size.

The second should fetch one of those blocks - with everything needed to start the next step, e.g. sending it to the DA committee. It should return as close to immediately as it takes to send the response.

The third won't return immediately, not if it's called too fast, because it only starts computing the VID precompile after it builds the block. (note: we may even change it to start computing after the second API request is received)

What this did after the loop wasn't great, but at least it didn't bury the discreet parts even one level of coupling deeper...

**Review by nyospe — COMMENTED** [review]
Thank you for getting rid of the loop.

I'm glad that part got done, but I'm not thrilled with where this went...

**Comment by nyospe** [comment]
So the issue here is, you set out to "simplify interaction with the builder", but the interaction had already been oversimplified.

File: crates/hotshot/src/lib.rs
Comment (jbearer) [reply crates/hotshot/src/lib.rs:693]: Oop, me too 🙊

File: crates/task-impls/src/transactions.rs
Comment (jbearer) [reply crates/task-impls/src/transactions.rs:264]: I'm not sure the "burying one level deeper" is really a problem. The only reason for having a separate function there is to make the retry loop a little simpler. We would want something along these lines regardless, even if it eventually only does the first API call in the inner loop. Other than that, the logic here is exactly the same as before except we won't call the same builder multiple times until we get a certain number of transactions or whatever.

I totally understand what you're saying about the correct way to use the API, and that would absolutely be an improvement. But, that's a bigger change to consensus than I want to make right now (especially the starting DA before VID part) and I'm inclined to go ahead with something along these lines anyways, because
* it will eliminate a variable when debugging the builder (problems arising due to repeat requests, plus this has slightly better timeout handling)
* it will simplify the config down in the sequencer by removing some parameters

**Review by nyospe — APPROVED** [review]
Okay. It'll need refactoring later, but this does solve the immediate issue.

File: crates/task-impls/src/builder.rs
Comment (QuentinI) [inline crates/task-impls/src/builder.rs:77]: Don't we want it to be equal to `builder_timeout`?

Comment (jbearer) [reply crates/task-impls/src/builder.rs:77]: I was thinking the client shouldn't know about consensus parameters, and this would just be a general "a simple HTTP request should never take longer than 2s or something is wrong"

Comment (jbearer) [reply crates/task-impls/src/builder.rs:77]: I think it's fine because we have the `builder_timeout` logic in consensus anyways

**Pull request merged**
